frontend/src/app/App.test.tsx

```diff
@@ -230,311 +237,378 @@ describe("App", () => {
+        instance.hostCommunicationMgr.props,
+        "onPageChange"
+      )
 
-    expect(wrapper.find("WithTheme(StatusWidget)").exists()).toBe(false)
-    expect(wrapper.find("ToolbarActions").exists()).toBe(false)
-  })
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "REQUEST_PAGE_CHANGE",
+            pageScriptHash: "hash1",
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-  it("shows the top bar if hideTopBar === false", () => {
-    const wrapper = shallow(<App {...getProps()} />)
-    wrapper.setState({ hideTopBar: false })
+      expect(onPageChangeFunc).toHaveBeenCalledWith("hash1")
+    })
 
-    expect(wrapper.find("WithTheme(StatusWidget)").exists()).toBe(true)
-    expect(wrapper.find("ToolbarActions").exists()).toBe(true)
-  })
+    it("shows hostMenuItems", () => {
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "SET_MENU_ITEMS",
+            items: [{ type: "separator" }],
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-  it("updates state.appPages when it receives a PagesChanged msg", () => {
-    const appPages = [
-      { icon: "", pageName: "bob", scriptPath: "bob.py" },
-      { icon: "", pageName: "carl", scriptPath: "carl.py" },
-    ]
+      // TODO: double check Main Menu re-renders based on prop change
+      wrapper.setState({ hideTopBar: false })
 
-    const msg = new ForwardMsg()
-    msg.pagesChanged = new PagesChanged({ appPages })
+      expect(instance.hostCommunicationMgr.state.menuItems).toStrictEqual([
+        { type: "separator" },
+      ])
+      expect(wrapper.find(MainMenu).prop("hostMenuItems")).toStrictEqual([
+        { type: "separator" },
+      ])
+    })
 
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
-    expect(wrapper.find("AppView").prop("appPages")).toEqual([])
+    it("shows hostToolbarItems", () => {
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "SET_TOOLBAR_ITEMS",
+            items: [
+              {
+                key: "favorite",
+                icon: "star.svg",
+              },
+            ],
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-    const instance = wrapper.instance() as App
-    instance.handleMessage(msg)
-    expect(wrapper.find("AppView").prop("appPages")).toEqual(appPages)
+      wrapper.setState({ hideTopBar: false })
+      expect(
+        wrapper.find(ToolbarActions).prop("hostToolbarItems")
+      ).toStrictEqual([
+        {
+          key: "favorite",
+          icon: "star.svg",
+        },
+      ])
+    })
+
+    // TODO: Fix this test
```

Comment:
What needs fixing? Can we fix it now?

Reply:
Just forgot to delete comment after test was fixed 😅